docs/pre-commit-hook-setup.md

```diff
@@ -0,0 +1,13 @@
+# CodeQL pre-commit-hook setup
+
+As stated in [CONTRIBUTING](../CONTRIBUTING.md) all CodeQL files must be formatted according to our [CodeQL style guide](ql-style-guide.md). You can use our pre-commit hook to avoid committing incorrectly formatted code. To use it, simply copy the [pre-commit](../misc/scripts/pre-commit) script to `.git/modules/ql/hooks/pre-commit` and make sure that it is executable.
```

Comment:
I left it out precisely because I didn't know what to write for Windows, yeah.

I can write "On Linux and macOS this can be done using chmod +x" and just leave out Windows altogether. If someone has a Windows machine and are willing to test if/how well it works then they can update the docs at a later point. How does that sound?